### PR TITLE
DIFM Lite: Adds a field to the site options API and changes to the admin menu

### PR DESCRIPTION
**Pull request by aneeshd16**
Summary:
P2: pdh1Xd-4l-p2
* Adds the `is_difm_lite_in_progress` to the site options API. (`/me/sites`). This option will return true if the `difm-lite-in-progress` sticker is present on the site. The sticker indicates that the DIFM Lite team is currently building the site. This option will be used in Calypso to block parts of the UI.
* Changes to the admin menu - If the `difm-lite-in-progress` sticker is present on the site, return the menu used for domain-only sites. This menu is restrictive and only contains the "Settings" menu.

Test Plan:
Optional - Test this with [#56935](https://github.com/Automattic/wp-calypso/pull/56935)

-----

* Sandbox [private link]
* Open the network console and log in to Calypso.
* Search for `https://[private link].2/me/sites` in the Network tab and check the response.
* The `options` object should have a new key: `is_difm_lite_in_progress`.
* Toggle the `difm-lite-in-progress` sticker via the Blog RC and verify that the value of the key is correctly returned in the API.

-----

* To test the admin menu, sandbox [private link] and log in to Calypso.
* Switch to a site with the `difm-lite-in-progress` sticker enabled and check the admin-menu (Left side)
{F9133796}

Reviewers: dzver, dehan91

Reviewed By: dzver

Tags: #touches_jetpack_files

Differential Revision: D68203-code

This commit syncs r233374-wpcom.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Please see D68203-code

### CODE REVIEW

**Comment by jeherve** [comment]
@aneeshd16 Did you have the chance to take a look at this?

Thanks!

File: projects/plugins/jetpack/modules/masterbar/admin-menu/load.php [hunk 1]
```diff
@@ -61,6 +61,13 @@ function get_admin_menu_class() {
 			return Domain_Only_Admin_Menu::class;
 		}
 
+		// DIFM Lite In Progress Sites. Uses the same menu used for domain-only sites.
+		$is_difm_lite_in_progress = has_blog_sticker( 'difm-lite-in-progress' );
```
[the remark this answers is not in the document]
Comment (aneeshd16) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/load.php:65]: If it's a WoA site, it looks like this function would return early in L49 above. And the condition in L53 already checks if it's a simple site. So we should be good here, right? 🙂

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 2]
```diff
@@ -669,6 +669,9 @@ protected function render_option_keys( &$options_response_keys ) {
 				case 'anchor_podcast':
 					$options[ $key ] = $site->get_anchor_podcast();
 					break;
+				case 'is_difm_lite_in_progress':
```
[the remark this answers is not in the document]
Comment (aneeshd16) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php:672]: I added this in D68203-code, I'm not sure how this got missed. 🤔

File: projects/plugins/jetpack/modules/masterbar/admin-menu/load.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeherve) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/load.php:67]: Yes you're right, I overlooked that, sorry!

**Review by jeherve — APPROVED** [review]
Should be good to merge now! 🚢

**Pull request merged**